share/namespace.go

```diff
@@ -90,7 +92,11 @@ func (n Namespace) ID() []byte {
 	return n.data[NamespaceVersionSize:]
 }
 
-func validate(version uint8, id []byte) error {
+// IsValidNamespace returns an error if the provided version is not
+// supported or the provided id does not meet the requirements
+// for the provided version. This should be user for validating
+// user specified namespaces
+func IsValidNamespace(version uint8, id []byte) error {
```

Comment:
My 2c: `IsXXX` or `IsValidXXX` functions mostly returns `bool`, but when the returned param is `error` `ValidateXXX` is more common.

Reply:
Fair point

Reply:
Not sure do we need `User` in the func name. I might missing some context, so feel free to resolve the comment.